src/components/tooltip/Tooltip.tsx

```diff
@@ -130,12 +130,21 @@ class Tooltip extends React.Component<TooltipProps, State> {
         this.setState({ hasRendered: true });
     }
 
-    componentDidUpdate(prevProps: TooltipProps) {
+    componentDidUpdate(prevProps: TooltipProps, prevState: State) {
+        const isControlled = this.isControlled();
+
         // Reset wasClosedByUser state when isShown transitions from false to true
-        if (this.isControlled()) {
+        if (isControlled) {
             if (!prevProps.isShown && this.props.isShown) {
                 this.setState({ wasClosedByUser: false });
             }
+        } else {
+            if (!prevState.isShown && this.state.isShown) {
+                document.addEventListener('keydown', this.handleKeyDown);
+            }
+            if (prevState.isShown && !this.state.isShown) {
+                document.removeEventListener('keydown', this.handleKeyDown);
```

Comment:
May need to remove the listener in componentWillUnmount as well, so it doesn't create dangling listeners